### DISCUSSION TITLE
Can I use private module for parsec?

### DISCUSSION THREAD

**Discussion by gihyeonjeon** [gem5/gem5#879] [Q&A] [answered]
HI! I'm studying about parsec-benchmarks.
I want to use the SimpleCache from gem5 tutorial for parsec benchmark.
But when I see the parsec benchmark, there are many codes need for that, such as sequencer, Port, System ...
Is it possible to learn how to write these codes through a tutorial or can you offer any advice on this?

**Comment by BobbyRBruce** [comment, marked as answer]
The big different between the Learning gem5 tutorial you're looking at and what's needed to run PARSEC is that tutorial uses SE-mode. You can almost certainly use something like "SimpleCache" for Parsec but first you need to get it working an FS-mode simulation.

Complete the Learning gem5 tutorial and then look at the tutorial on using the stdlib: https://www.gem5.org/documentation/gem5-stdlib/overview. I suspect once you get to this point you'll understand more and be more confident of what you need to do to get the simulation you want.

A recent tutorial we did at HPCA may be of interest to you. See here: https://www.gem5.org/events/hpca-2024
